protocol/syncer.go

```diff
@@ -296,7 +306,8 @@ func (s *Syncer) Start() {
 func (s *Syncer) BestPeer() *syncPeer {
 	var bestPeer *syncPeer
 	var bestTd *big.Int
-
+	s.peersLock.RLock()
+	defer s.peersLock.RUnlock()
```

Comment:
Please move `RUnlock()` call to ~L316, just after the for loop. (no need for `defer` then)

Reply:
Irrelevant after commit 9593ada